src/adapters/index.js

```diff
@@ -0,0 +1,30 @@
+import * as chirpstack from './chirpstack.js';
+import errors from '../errors.js';
+
+const adapters = {
+    chirpstack
+};
+
+// fastify force all headers in lowercase
+const headerName = 'ns-adapter';
+
+
+export default function adaptData ( request, reply ) {
+    const adapterName = request.headers[headerName];
+
+    if ( !adapterName && adapters[adapterName] ) {
+        reply.sendError(errors.BAD_REQUEST, `Unsupported adapter "${adapterName}"`);
+
+        return false;
+    }
+
+    const adapter = adapters[adapterName];
+    const body = adapter(request, this.log);
+
+    if ( !body ) {
```

Comment:
You don't use the adapter function result

Reply:
fixed